### PR TITLE
Spread enum

### LINKED ISSUE #591: Support extending enums through spread
Implementation issue for #395

Allow spread to bring the items from one enumeration into another.  The result is a new enum with no relationship with the previous enum.

Example:
```
enum bar {
  One: "1", 
  Two: "2", 
  Three: "3"
}
enum foo  {
  ...bar;
  Four: "4", 
  Five: "5", 
  Six: "6"
}
```

enum `foo` will have 6 elements, with ```bar.One !== foo.One```

### PR DESCRIPTION
**Pull request by leni-msft**
fixes #591

### CODE REVIEW

File: packages/compiler/formatter/print/printer.ts [hunk 1]
```diff
@@ -460,6 +463,14 @@ export function printEnumMember(
   return [multiline && isNotFirst ? hardline : "", decorators, id, value];
 }
 
+function printEnumSpreadMember(
```
Comment (timotheeguerin) [inline packages/compiler/formatter/print/printer.ts:466]: Add test for the formatter

File: packages/compiler/core/checker.ts [hunk 2]
```diff
@@ -2573,6 +2580,44 @@ export function createChecker(program: Program): Checker {
...
+            createDiagnostic({
+              code: "enum-member-duplicate",
+              format: { name: member.name },
+              target: targetNode,
+            })
+          );
+        } else {
+          existingMemberNames.add(member.name);
+          const clonedMember = cloneType(member, {
+            enum: parentEnum,
```
Comment (timotheeguerin) [inline packages/compiler/core/checker.ts:2609]: We might want to have a `sourceMember` like we have `sourceProperty` for spread properties

File: packages/compiler/core/parser.ts [hunk 3]
```diff
@@ -26,6 +26,7 @@ import {
   DirectiveExpressionNode,
```
Comment (timotheeguerin) [inline packages/compiler/core/parser.ts:26]: Believe that might require a change to `spec.enum.html`? @nguerrera

Comment (leni-msft) [reply packages/compiler/core/parser.ts:26]: thanks, updated

File: packages/compiler/formatter/print/printer.ts (hunk 1 above)
Comment (leni-msft) [reply packages/compiler/formatter/print/printer.ts:466]: done

**Review by timotheeguerin — APPROVED** [review]

**Pull request merged**